### PR TITLE
Fix fork tests on Windows.

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: src/pipe.toit [hunk 1]
```diff
@@ -383,6 +383,7 @@ Forks a program, and returns the exit status.
 A return value of zero indicates the program ran without errors.
 Uses the /bin/sh shell to parse the command, which is a single string.
 Arguments are split by the shell at unescaped whitespace.
+  On Windows we just split at spaces, since the shell is not available.
```
Comment (floitsch) [inline src/pipe.toit:386]: Wouldn't it be nicer to try to be a bit smarter? Like `system "\"foo bar\""` should probably not be split.

**Review by floitsch — APPROVED** [review]
Reluctant LGTM.
I feel this doesn't really "fix" tests on Windows, but mostly disables them.

File: src/pipe.toit (hunk 1 above)
Comment (erikcorry) [reply src/pipe.toit:386]: I think it's not worth it.  Added comment to direct users to run_program for non-trivial cases.

File: src/pipe.toit [hunk 2]
```diff
@@ -391,7 +392,7 @@ The $environment argument is used as in $fork.
 */
 system --environment/Map?=null command -> int?:
   if platform == PLATFORM_WINDOWS:
-    return run_program --environment=environment ["cmd", "/s", "/c",  command]
+    return run_program --environment=environment ["cmd", "/s", "/c"] + (command.split " ")
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply src/pipe.toit:395]: I don't see how that would help.

File: tests/pipe_test_slow.toit [hunk 3]
```diff
@@ -184,8 +185,14 @@ main args:
...
 
+/// Returns whether a path exists and is a character device.
+is_char_device name --follow_links/bool=true -> bool:
+  stat := file.stat name --follow_links
+  if not stat: return false
+  return stat[file.ST_TYPE] == file.CHARACTER_DEVICE
+
 pipe_large_file:
-  if (file.is_file "/bin/sh") and (file.is_file "/usr/bin/cat"):
+  if (file.is_file "/bin/sh") and (file.is_file "/usr/bin/cat") and (is_char_device "/dev/null"):
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply tests/pipe_test_slow.toit:195]: Yes.

**Pull request merged**